columnflow/tasks/histograms.py

```diff
@@ -243,8 +247,17 @@ def expr(events, *args, **kwargs):
                                 if len(events) == 0 and not has_ak_column(events, route):
                                     return empty_f32
                                 return route.apply(events, null_value=variable_inst.null_value)
+                        arr = expr(events)
+                        # prepare the selection
+                        sel = variable_inst.selection
+                        if sel != "1":
+                            if callable(sel):
+                                mask = sel(events)
+                                arr = ak.where(mask, arr, variable_inst.null_value)
+                            else:
+                                raise ValueError(f"invalid selection: {sel}")
```

Comment:
Just some de-nesting.

```suggestion
                            if not callable(sel):
                                raise ValueError(f"invalid selection: {sel}")
                            mask = sel(events)
                            arr = ak.where(mask, arr, variable_inst.null_value)
```